### ISSUE TITLE
Spark transformer params missing after deserializing

### ISSUE THREAD

**Issue by pnpritchard** [combust/mleap#40]
I have started experimenting with serializing and deserializing spark pipelines. However, I have noticed that when I override default params, they are missing after deserialization. I have narrowed down the cause to how the `OpNode#load` method is implemented, and specifically the use of `.copy(model.extractParamMap())`.

I am not very familiar with this `copy` API provided by spark, so cannot figure out if this is a Spark bug or if this is misuse of the API. So the only solution I've thought so far is just to explicitly get and set each param (as done in `OpModel#load`).

Here is a reproducible case, using `Binarizer` as an example:
```scala
import org.apache.spark.ml.feature.Binarizer
import ml.combust.mleap.spark.SparkSupport._

val bin = new Binarizer("bin")
  .setInputCol("in")
  .setOutputCol("out")
  .setThreshold(0.5)

val path = new File(...)
bin.serializeToBundle(path)

val bin2 = path.deserializeBundle()._2.asInstanceOf[Binarizer]
assert(bin.getInputCol == bin2.getInputCol)
assert(bin.getOutputCol == bin2.getOutputCol)
assert(bin.getThreshold == bin2.getThreshold) //fails
```

**Comment by hollinwilkins** [comment]
Hey, great catch. Do you have a PR by any chance? If not, I will get on this in the next week.

**Comment by hollinwilkins** [comment]
I am going to fix this by using the pattern:

```scala
var transformer = ...
transformer = transformer.set(transformer.threshold, model.get(model.threshold))
```

This should account for the optional nature of a lot of these paramters

**Comment by hollinwilkins** [comment]
Adding in tests for verifying Spark deserialization has uncovered a few other errors, I'll be fixing these as well then putting in a PR. Will let you know when it is ready for review.

**Comment by hollinwilkins** [comment]
PR: https://github.com/combust-ml/mleap/pull/41 is ready for review

**Comment by pnpritchard** [comment]
Awesome, thanks! Fix looks good and the new tests are even better.

**Comment by hollinwilkins** [comment]
Great, this fix is in the current snapshot version 0.5.0-SNAPSHOT on OSS Sonatype. It will be included in the next version of MLeap 0.5.0 when we make a release in the next week or so. Thanks for tracking this issue down, and please let us know of any feature enhancements or bugs you find!
